lib/osf-components/addon/components/file-browser/add-new/upload/template.hbs

```diff
@@ -1,85 +1,110 @@
-                    </ul>
-                {{/if}}
-                {{#if this.shouldShowSuccessModal}}
-                    <ul local-class='FileList'>
-                        {{#each this.uploadCompleted as |item|}}
-                            <li
-                                local-class='FileListItem'
-                            >
-                                <div>
-                                    <FaIcon @icon='file-alt' />
-                                    <span local-class='Filename'>{{item.name}}</span>
-                                </div>
-                                <FaIcon local-class='CheckIcon' @icon='check' />
-                            </li>
-                        {{/each}}
-                    </ul>
-                {{/if}}
-            </dialog.main>
-        </OsfDialog>
-    </DropzoneWidget>
-{{/let}}
+                                    <div>
+                                        <FaIcon @icon='file-alt' />
+                                        <span local-class='Filename'>{{item.name}}</span>
+                                    </div>
+                                    <Button
+                                        @layout='fake-link'
+                                        {{on 'click' (fn this.retryUpload dropzone.dropzoneInstance item)}}
+                                    >
+                                        {{#if (or (eq item.status 'uploading') (eq item.status 'queued'))}}
+                                            <FaIcon @icon='spinner' @pulse={{true}} />
+                                        {{else}}
+                                            <FaIcon @icon='undo' />
+                                        {{/if}}
+                                    </Button>
+                                </li>
+                            {{/each}}
+                        </ul>
+                    {{/if}}
+                    {{#if this.shouldShowSuccessModal}}
+                        <ul local-class='FileList'>
+                            {{#each this.uploadCompleted as |item|}}
+                                <li
+                                    local-class='FileListItem'
+                                >
+                                    <div>
+                                        <FaIcon @icon='file-alt' />
+                                        <span local-class='Filename'>{{item.name}}</span>
+                                    </div>
+                                    <FaIcon local-class='CheckIcon' @icon='check' />
+                                </li>
+                            {{/each}}
+                        </ul>
+                    {{/if}}
+                </dialog.main>
+                <dialog.footer>
+                    {{#if this.shouldShowUploadingModal}}
+                        <Button
+                            {{on 'click' dropzone.dropzoneInstance.removeAllFiles}}
+                            @type='destroy'
```

Comment:
This being a destroy button doesn't quite feel right. Also, should this close the modal as well?

Reply:
This would cancel all upload in progress and will direct users to the failure page of the modal. So it shouldn't close the modal.